cmd/traffic/cmd/manager/internal/state/state.go

```diff
@@ -608,7 +628,18 @@ func (s *State) UpdateIntercept(interceptID string, apply func(*rpc.InterceptInf
 }
 
 func (s *State) RemoveIntercept(interceptID string) bool {
-	_, didDelete := s.intercepts.LoadAndDelete(interceptID)
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	return s.unlockedRemoveIntercept(interceptID)
+}
+
+func (s *State) unlockedRemoveIntercept(interceptID string) bool {
```

Comment:
Why is this method is prefixed with `unlocked` when it's called with the `s.mu.Lock()` in place?

Reply:
that seems to be the naming convention for the class. If the method is supposed to be called with the caller holding the mutex then it's `unlocked`. See `unlockedRemoveSession` or `unlockedCheckAgentsForIntercept` for other examples. Doesn't mean I agree mind you but I did wanna keep it consistent 🙃

Reply:
That is ... weird. But OK :-)